### ISSUE TITLE
Display type aliases in auto-import suggestions

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#606]
I really like the change to the auto-import suggestions (#163) introduced with the last release `2020.11.1`.
However it doesn't seem like type aliases get indexed. I've already set `python.analysis.indexing` to `true`.
Maybe this is a feature worth considering.

**Comment by jakebailey** [comment]
Can you clarify what you mean? Are you saying that the names themselves should appear as symbols in the workspace symbol search? We don't include these in document symbols (the outline) or workspace symbols (but I don't think we did before either).

**Comment by cdce8p** [comment]
@jakebailey Given two files:
```py
# const.py
MyAlias = list[int]
MyConst = 42

class MyClass:
    ...


# main.py
var: [Alias]
```

If I start writing `My` for `Alias` I will only get the suggestion for `MyClass`, although `MyAlias` would be equally valid.

--
As a side note, even `MyConst` isn't suggested. It seems only constants with all capital letters are indexed. If I change it to `MY_CONST`, it works.

**Comment by jakebailey** [comment]
Ah, my mistake. "type aliases", I took it to mean an import alias in `import X as Y` as that's part of what #163 was about.

With indexing enabled, you should get the completion for that (or, by opening up the file so it's loaded). Is it correct that you're saying it's missing?

**Comment by cdce8p** [comment]
Yeah, at least I though so. After some further investigation I found that `MyAlias` would **not** get indexed where as `MY_ALIAS` would. This is the same behavior as with constants. However, I believe type aliases commonly use CamelCase. https://www.python.org/dev/peps/pep-0484/#type-aliases

**Comment by jakebailey** [comment]
There's some heuristics to indexing to prevent it from taking too long in library code, but if this is all within user code, we should be doing a better job.

**Comment by heejaechang** [comment]
@cdce8p yes, you are right on the heuristics. since python doesn't have a formal way (like export in typescript) to specify which variables should be accessible from others, including every variable in the auto-import causes too much noise in completion. (easily cause hundreds of more entries to be inserted into completion).

so we have this [heuristics](https://github.com/microsoft/pyright/blob/master/packages/pyright-internal/src/languageService/autoImporter.ts#L285)

but that being said, if we can come up with better heuristics, we can try that. 

FYI @savannahostrowski  @judej

**Comment by erictraut** [comment]
I think it makes sense to include capitalized (camel-case) variables defined at the module level. These are likely type aliases.

Also, I noticed that your current heuristic uses a reg-ex that is incorrect because it doesn't allow numbers or underscores. See symbolNameUtils.isConstantName for a better approach.

**Comment by heejaechang** [comment]
@erictraut existing regExp is negative check, so it disallows any variable that contains non capital char, so it will allow number and such that is legal to identifier. It doesn't care about space and such since name already passed identifier name check.

but sure, I can change it to isConstantName if that is prefered.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.11.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020112-18-november-2020
